rpcs3/rpcs3qt/update_manager.cpp

```diff
@@ -543,18 +543,36 @@ bool update_manager::handle_rpcs3(const QByteArray& rpcs3_data, bool automatic)
 
 	replace_path = Emulator::GetEmuDir() + "rpcs3.exe";
 
+	// Creating a file to indicate we're restarting
+	const std::string s_filelock = fs::get_cache_dir() + ".restart_lock";
+	fs::file restart_file(s_filelock, fs::read + fs::rewrite);
+	restart_file.close();
+
 #endif
 
 	m_progress_dialog->close();
-
 	QMessageBox::information(m_parent, tr("Auto-updater"), tr("Update successful!"));
-	int ret = execl(replace_path.c_str(), replace_path.c_str(), nullptr);
 
+#ifdef __linux__
```

Comment:
Changes are kinda... "better is the enemy of good".
If execl worked, why change it?
And what'll happen on exception throw? Qt doesn't have any catch iirc.

Reply:
I may go back to execl, I'm trying to figure out what the actual issue is, the behaviour of execl on windows is a bit undefined(it doesn't replace the process but spawn a new one and then terminate the current one) so I thought I'd try doing it manually and ensuring quick process termination but that doesn't seem to be the issue.

Like the title says, WIP.